core/config/project_settings.cpp

```diff
@@ -1114,6 +1114,7 @@ ProjectSettings::ProjectSettings() {
 
 	// Keep the enum values in sync with the `DisplayServer::ScreenOrientation` enum.
 	custom_prop_info["display/window/handheld/orientation"] = PropertyInfo(Variant::INT, "display/window/handheld/orientation", PROPERTY_HINT_ENUM, "Landscape,Portrait,Reverse Landscape,Reverse Portrait,Sensor Landscape,Sensor Portrait,Sensor");
+	custom_prop_info["display/window/vsync/vsync_mode"] = PropertyInfo(Variant::STRING, "display/window/vsync/vsync_mode", PROPERTY_HINT_ENUM, "Disabled,Enabled,Adaptive,Mailbox");
```

Comment:
I just noticed this is a *string* enum rather than an integer enum. I think using an integer enum would be better as it's more Godot-y overall. (We're trying to move away from string usage when possible in 4.0 onwards.)

Reply:
I presume I just copied how it was done for the screen orientation (which was before you changed it to an integer enum). But you are right, an integer enum is cleaner as it wouldn't need that ugly if-elseif construct in main.cpp. I will open a PR soon to change that.